compiler/optimizer/OMRValuePropagation.cpp

```diff
@@ -4097,6 +4096,10 @@ void TR::GlobalValuePropagation::processStructure(TR_StructureSubGraphNode *node
       }
    else
       {
+      // Only commoned nodes within a block need to be set in _defMergedNodes.
+      // So it should be reset here
+      TR_ASSERT(!node->getBlock()->isExtensionOfPreviousBlock(), "This optimization does not run on extended blocks");
+      _defMergedNodes->empty();
```

Comment:
Would you mind using `TR_ASSERT_FATAL` here? I think that `TR_ASSERT` still doesn't get checked in most (or even all?) of the testing

Reply:
Yeah, I was wondering about that but all asserts in this file are non-fatal. Anyways, let me make it FATAL just to be sure.